TombEngine/Game/effects/effects.cpp

```diff
@@ -657,16 +657,61 @@ void TriggerExplosionSparks(int x, int y, int z, int extraTrig, int dynamic, int
 
 	if (uw == 1)
 	{
-		spark.sG = (GetRandomControl() & 0x3F) + 128;
-		spark.sB = 32;
-		spark.dR = 192;
-		spark.dG = (GetRandomControl() & 0x1F) + 64;
-		spark.dB = 0;
 		spark.colFadeSpeed = 7;
 		spark.fadeToBlack = 8;
 		spark.life = (GetRandomControl() & 7) + 16;
 		spark.sLife = spark.life;
 		spark.roomNumber = roomNumber;
+
+		if (mainColor == Vector3::Zero)
+		{
+		spark.sG = (GetRandomControl() & 0x3F) + 128;
+		spark.sB = 32;
+		spark.dR = 192;
+		spark.dG = (GetRandomControl() & 0x1F) + 64;
+		spark.dB = 0;
+		}
+		else
+		{
+			// New colored flame processing.
```

Comment:
Why this comment is saying something about "colored flame"? It indicates unchecked copypaste.